src/dune_pkg/sys_poll.ml

```diff
@@ -50,9 +50,13 @@ let arch ~path =
     match Sys.os_type with
     | "Unix" | "Cygwin" -> uname ~path [ "-m" ]
     | "Win32" ->
-      if Sys.word_size = 32 && not (OpamStubs.isWoW64 ())
-      then Fiber.return @@ Some "i686"
-      else Fiber.return @@ Some "x86_64"
+      (match OpamStubs.getArchitecture () with
```

Comment:
Because all the clauses do `Fiber.return`, you can factor it out to the outside:

```ocaml
Fiber.return @@ match ..
```

Reply:
Indeed, I didn't see the forest for the trees.